ext/standard/array.c

```diff
@@ -2160,7 +2161,7 @@ PHP_FUNCTION(range)
 			RANGE_CHECK_DOUBLE_INIT_ARRAY(low, high);
 
 			ZEND_HASH_FILL_PACKED(Z_ARRVAL_P(return_value)) {
-				for (i = 0, element = low; i < size && element >= high; ++i, element = low - (i * step)) {
+				for (i = 0, element = low; i < size && element >= (high - DBL_EPSILON); ++i, element = low - (i * step)) {
```

Comment:
A more correct approach of this check would look in pseudo code similar to `abs(a - b) < threshold`.

Thanks.

Reply:
@weltling, thanks for the feedback.

I've moved the constant to the LHS as it makes it more readable. Let me know if you prefer it differently.